player/javascript.c

```diff
@@ -836,6 +836,59 @@ static void script_readdir(js_State *J, void *af)
     }
 }
 
+static void script_stat(js_State *J)
+{
+    const char *path = js_tostring(J, 1);
+
+    struct stat statbuf;
+    if (stat(path, &statbuf) != 0) {
+        push_failure(J, "Cannot stat path");
+        return;
+    }
+    // Clear last error
+    set_last_error(jctx(J), 0, NULL);
```

Comment:
Hmm.. the error is not cleared on other functions IIRC. Can you please check whether it should be applied in more places? if yes, we should probably put it into a function `clear_last_error()` or some such. It could be a followup too, but I'd still appreciate if you could check it out.

Reply:
OK, it's correct here, so let's leave it. I'll review all the functions which set last error and make sure all of them indeed set it on all cases. When I think of it, they probably do, but I'll re-verify it nevertheless.

Reply:
So I just went over all the APIs which are documented to set last error, and indeed all of them set it on all code paths. readdir is indeed the only exception which clears it first, because it doesn't use one of the utils which set it automatically like the others do (push_status, pushed_error, etc).

And stat will be another such exception. That's OK.